geoengine/workflow.py

```diff
@@ -520,11 +521,17 @@ def save_as_dataset(
         session = get_session()
 
         request_body = {
-            'name': name,
+            'displayName': display_name,
             'description': description,
             'query': query_rectangle,
         }
 
+        if name is not None:
+            request_body['name'] = name
+
+        if len(display_name) == 0 and name is not None:
```

Comment:
if you really want to allow specifying no display name, why not make it Optional in the function parameters?

Reply:
[9a5782d](https://github.com/geo-engine/geoengine-python/pull/141/commits/9a5782d7a434f788de682b06d5d61512a8a50e44)

Reply:
why did you not make it optional instead?

Reply:
i think it is strange when both are optional